### DISCUSSION TITLE
Could we have a .deb ?

### DISCUSSION THREAD

**Discussion by phenix-factory** [qutebrowser/qutebrowser#6444] [General] [answered]
Hello !

I would love to use Qutebrowser, but the version inside the repositories is 1.1 and look never updated.
Installing it with python is quite painful and does not integrate well with the desktop.

I don't know if it's hard to do, maybe it doesn't worth it ?

**Comment by The-Compiler** [comment, marked as answer]
"The repositories"? Which repositories exactly? Debian has a (mostly) [up-to-date package](https://packages.debian.org/sid/qutebrowser), which you might or might not be able to install on other Debian-based systems as well.

**Reply by phenix-factory** [reply]
I'm using Linux Mint, so it's ubuntu 20.04 repositories.
Would be nice to have qutebrowser update like all other apps when you make a new release. But again, it may be a lot of works.

**Reply by The-Compiler** [reply]
Those Debian packages are usually up-to-date. I'll check back with @fiete201 why they've fallen behind a bit.

As for Linux Mint (or Ubuntu 20.04), that's how those LTS distributions work: You don't get updates for applications (other than critical bug- and security fixes) until you upgrade your entire distribution. I don't see what you mean with "like all other apps".

In any case, note that Ubuntu 20.04 also ships an outdated version [of the underlying QtWebEngine](https://packages.ubuntu.com/focal/libqt5webengine5), so you're missing out on [about a year](https://wiki.qt.io/Qt_5.12_Release) of security fixes (QtWebEngine in Debian/Ubuntu doesn't get security fixes [at all](https://www.debian.org/releases/stable/amd64/release-notes/ch-information.en.html#browser-security)).

**Comment by phenix-factory** [comment]
> As for Linux Mint (or Ubuntu 20.04), that's how those LTS
> distributions work: You don't get updates for applications (other than
> critical bug- and security fixes) until you upgrade your entire
> distribution.

Firefox is up to date. And I can add repositories/ppa for apps that
provide them.

>I don't see what you mean with "like all other apps".

I just mean by APT update/upgrade.

> In any case, note that Ubuntu 20.04 also ships an outdated version [of
> the underlying
> QtWebEngine](https://packages.ubuntu.com/focal/libqt5webengine5), so
> you're missing out on about a year of security fixes (QtWebEngine in
> Debian/Ubuntu doesn't get security fixes [at
> all](https://www.debian.org/releases/stable/amd64/release-notes/ch-information.en.html#browser-security)).

Thanks for sharing this, now, I know qutebrowser can't be used on my
system. I m sad...

**Reply by The-Compiler** [reply]
> Firefox is up to date. And I can add repositories/ppa for apps that provide them.

Firefox is a rare exception. Sure, with PPAs you get updates, but those just "circumvent" your normal distribution packaging. For anything outside of Chromium/Firefox or things you installed via PPA (or e.g. Snap), your distribution will not keep you up to date.

> Thanks for sharing this, now, I know qutebrowser can't be used on my system. I m sad...

You could always [install via a virtualenv](https://github.com/qutebrowser/qutebrowser/blob/master/doc/install.asciidoc#tox) which gives you a (mostly up-to-date Qt as a binary.

**Comment by phenix-factory** [comment]
>> Thanks for sharing this, now, I know qutebrowser can't be used on my system. I m sad...
>
> You could always [install via a virtualenv](https://github.com/qutebrowser/qutebrowser/blob/master/doc/install.asciidoc#tox) which gives you a (mostly up-to-date Qt as a binary.

Yeah, but it does integrate well with the desktop, I don't think you can
make it the default browser ?
I guess I could automate the update with a daily cron.

I get it's not that hard to do and I m just lazy, but I really think
having a .deb file could help some people to install qutebrowser.

Any way, thanks for your time and work !

**Reply by The-Compiler** [reply]
> Yeah, but it does integrate well with the desktop, I don't think you can make it the default browser ?

You can probably download the [`.desktop` file](https://github.com/qutebrowser/qutebrowser/blob/master/misc/org.qutebrowser.qutebrowser.desktop) and put it into `~/.local/share/applications`.

> I get it's not that hard to do and I m just lazy, but I really think having a .deb file could help some people to install qutebrowser.

There is a `.deb` file from Debian, see my earlier link. It looks like you want a PPA, not a `.deb` file. A proper PPA would likely also want to ship updated versions of Qt/QtWebEngine, and doing that is a lot of work.
